### PR TITLE
embed: fix revision-based compaction with default value

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/9337.

### CODE REVIEW

**Comment by fanminshi** [comment]
taking a look!

File: embed/config_test.go [hunk 1]
```diff
@@ -128,6 +128,15 @@ func TestUpdateDefaultClusterFromNameOverwrite(t *testing.T) {
 	}
 }
 
+func TestInvalidAutoCompactionMode(t *testing.T) {
```
Comment (fanminshi) [inline embed/config_test.go]: maybe add another test to the  issue scenario:
```go
       cfg := NewConfig()
       cfg.AutoCompactionMode = "revision"
       cfg.AutoCompactionRetention = "1"
       srv, err := embed.StartEtcd(cfg)
       // verify that srv.Config().AutoCompactionRetention == time.Duration(1)
```

**Comment by fanminshi** [comment]
lgtm

**Comment by xiang90** [comment]
lgtm

**Pull request merged**